feat(service-worker): remove deprecated `versionedFiles` option

alan-agius4 opened the pull request:
BREAKING CHANGE:

Remove deprecated option `versionedFiles` from service worker asset group configuration in `ngsw-config.json`

Before
```json
"assetGroups": [
  {
    "name": "test",
    "resources": {
      "versionedFiles": [
        "/**/*.txt"
      ]
    }
  }
]
```

After
```json
"assetGroups": [
  {
    "name": "test",
    "resources": {
      "files": [
        "/**/*.txt"
      ]
    }
  }
]
```

gkalpak commented on aio/content/guide/service-worker-config.md:
```diff
@@ -108,8 +106,6 @@ This section describes the resources to cache, broken up into three groups.
 
```
The [sentence][1] right above this line also needs to be updated:

```diff
-This section describes the resources to cache, broken up into three groups.
+This section describes the resources to cache, broken up into the following groups:
```

[1]: https://github.com/angular/angular/blame/a1a3c336e/aio/content/guide/service-worker-config.md#L105

alan-agius4 replied:
updated

gkalpak commented on packages/service-worker/config/src/generator.ts:
```diff
@@ -45,26 +45,14 @@ export class Generator {
       const allFiles = await this.fs.list('/');
 
       const plainFiles = allFiles.filter(fileMatcher).filter(file => !seenMap.has(file));
       plainFiles.forEach(file => seenMap.add(file));
```
Now that `versionedFiles` is gone, `plainFiles` doesn't make sense as a variable name.
I would go with `matchedFiles` instead.

alan-agius4 replied:
updated

gkalpak commented on packages/service-worker/config/src/generator.ts:
```diff
@@ -45,26 +45,14 @@ export class Generator {
-
       // Add the hashes.
-      const matchedFiles = [...plainFiles, ...versionedFiles].sort();
+      const matchedFiles = plainFiles.sort();
```
Now that we don't need to merge, I would move the `sort()`-ing to the line where `plainFiles` is initialized:

```diff
-const plainFiles = allFiles.filter(fileMatcher).filter(file => !seenMap.has(file));
+const plainFiles = allFiles.filter(fileMatcher).filter(file => !seenMap.has(file)).sort();
```

<sub>And then I would renamed `plainFiles` to `matchedFiles` :wink:</sub>

gkalpak reviewed (changes requested):
Awesome, thx, @alan-agius4 ✨
I am not sure how we handle removals like that. Questions I have:
- Is `feat` the right type for the commit message.
- Don't we need a `BREAKING CHANGE` notice in the commit message (and the changelog).

☝️ @IgorMinar?

alan-agius4 commented:
I checked a couple of commits removing deprecation is sometimes typed with `fix` and sometimes with `feat`. Also sometimes it's marked a breaking change and sometimes not.

I added the `BREAKING CHANGE` footer as I think it makes sense. Please let me know if I should change the commit type.

gkalpak reviewed (approved):
Thx for adding the breaking change notice. A couple of nits:

- According to our [guidelines][1], `BREAKING CHANGE` should be followed by a `:` and two newlines.

- I would add some more info regarding _where_ `versionedFiles` was removed from (from a user's perspective):<br />
  _"Remove deprecated option \`versionedFiles\` from service worker **asset group configuration in \`ngsw-config.json\`**."_

- Specify the language (`json`) in code snippets: `` ``` `` --> `` ```json ``

-  Typos/Nits:
    - `'test'` --> `"test"`
    - Avoid trailing commas (which are invalid in JSON).
    - Use consistent indentation 😇 

[1]: https://github.com/angular/angular/blob/e9867e8ae12bec/CONTRIBUTING.md#footer

alan-agius4 commented:
@gkalpak, thanks for the review updated comments

alan-agius4 committed:
feat(service-worker): remove deprecated `versionedFiles` option

BREAKING CHANGE:

Remove deprecated option `versionedFiles` from service worker asset group configuration in `ngsw-config.json`

Before
```json
"assetGroups": [
  {
    "name": "test",
    "resources": {
      "versionedFiles": [
        "/**/*.txt"
      ]
    }
  }
]
```

After
```json
"assetGroups": [
  {
    "name": "test",
    "resources": {
      "files": [
        "/**/*.txt"
      ]
    }
  }
]
```

atscott commented:
Hey Alan, should you get Igor to review this or not necessary?

alan-agius4 commented:
Lets bounce it of with @IgorMinar

IgorMinar commented on aio/content/guide/deprecations.md:
```diff
@@ -211,14 +210,6 @@ Support for using the `ngModel` input property and `ngModelChange` event with re
 For more information, see the usage notes for [`FormControlDirective`](api/forms/FormControlDirective#use-with-ngmodel) and [`FormControlName`](api/forms/FormControlName#use-with-ngmodel).
 
 
-{@a sw-versionedfiles}
```
instead of completely removing this, can you moved this into the "removed" section instead: https://angular.io/guide/deprecations#removed-apis thanks!

IgorMinar reviewed (approved):
otherwise lgtm. thanks

mary-poppins commented:
You can preview dd811ce at https://pr32862-dd811ce.ngbuilds.io/.
You can preview a1a3c33 at https://pr32862-a1a3c33.ngbuilds.io/.
You can preview 36b44f2 at https://pr32862-36b44f2.ngbuilds.io/.
You can preview edae44b at https://pr32862-edae44b.ngbuilds.io/.

mary-poppins commented:
You can preview 37bb3c0 at https://pr32862-37bb3c0.ngbuilds.io/.

alan-agius4 committed:
fixup! feat(service-worker): remove deprecated `versionedFiles` option
